{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "037de0da-cca0-417b-aec3-17f7d2fe7726",
   "metadata": {},
   "source": [
    "## Analyze ClinVar UDN Submissions\n",
    "\n",
    "### Description\n",
    "A request was made in the [Mondo repo](https://github.com/monarch-initiative/mondo/issues/7773) to ensure ClinVar submissions from UDN are in Mondo. This notebook is intended to filter out the known positives (easily identified diseases) from the list (albeit knowing there would be some false positives) to scope out the curation effort.\n",
    "\n",
    "### Resources\n",
    "- ClinVar UDN https://www.ncbi.nlm.nih.gov/clinvar/submitters/505999/\n",
    "- See [Total submissions link](https://www.ncbi.nlm.nih.gov/clinvar/?LinkName=orgtrack_clinvar&from_uid=505999) for all submissions and Download file link.\n",
    "\n",
    "### Summary\n",
    "See file `/data/output/clinvar_udn_submissions-no_mondo.tsv`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a4d09c9-9819-4c2a-8d52-b57bc102f9ef",
   "metadata": {},
   "source": [
    "### Install Python packages needed for this notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c658a712-af63-4965-be6c-fbb35375ed1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Install Python packages\n",
    "# !pip install oaklib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "40fd6bae-3242-4985-b521-7d5ed1c67fc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Test OAK Installation\n",
    "# !runoak --help"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b9af29a-aecd-4da0-b4e9-92512a176b46",
   "metadata": {},
   "source": [
    "### Load Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c032e3a9-cc4b-4269-b016-1f695236a5c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# imports\n",
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "from oaklib import get_adapter\n",
    "from tqdm.notebook import tqdm\n",
    "\n",
    "# Configure dataframe display\n",
    "pd.set_option('display.max_columns', None)\n",
    "# pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_colwidth', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2462245a-2d5c-4bc8-a09e-ef4b8a641630",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a wrapper to connect to the mondo database\n",
    "adapter = get_adapter(f\"sqlite:obo:mondo\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff286321-b269-4086-aa69-42e5a00a311b",
   "metadata": {},
   "source": [
    "### Check the database connection and get the ontology metadata and check ontology version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "72cb9025-e4e0-49a3-9429-251b3bd11fb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': ['obo:mondo.owl'],\n",
       " 'IAO:0000700': ['MONDO:0042489',\n",
       "  'MONDO:0021178',\n",
       "  'MONDO:0021125',\n",
       "  'MONDO:0000001'],\n",
       " 'dce:description': ['A semi-automatically constructed ontology that merges in multiple disease resources to yield a coherent merged ontology.'],\n",
       " 'dce:title': ['Mondo Disease Ontology'],\n",
       " 'dcterms:license': ['cc:by/4.0/'],\n",
       " 'dcterms:source': ['<https://rarediseases.info.nih.gov/>',\n",
       "  '<https://id.nlm.nih.gov/mesh/>',\n",
       "  '<http://www.orpha.net/ontology/orphanet.owl>',\n",
       "  'obo:uberon.owl',\n",
       "  'obo:ncit.owl',\n",
       "  'obo:ncbitaxon.owl',\n",
       "  'obo:mf.owl',\n",
       "  'obo:hp.owl',\n",
       "  'obo:go.owl',\n",
       "  'obo:envo.owl',\n",
       "  'obo:doid.owl',\n",
       "  'obo:chebi.owl'],\n",
       " 'foaf:homepage': ['http://obofoundry.org/ontology/mondo.html'],\n",
       " 'oio:hasOBOFormatVersion': ['1.2'],\n",
       " 'owl:versionIRI': ['obo:mondo/releases/2024-06-04/mondo.owl'],\n",
       " 'rdf:type': ['owl:Ontology'],\n",
       " 'rdfs:comment': ['Includes Ontology(OntologyID(Anonymous-11)) [Axioms: 72899 Logical Axioms: 0]'],\n",
       " 'sh:prefix': ['obo'],\n",
       " 'schema:url': ['http://purl.obolibrary.org/obo/mondo.owl'],\n",
       " 'rdfs:isDefinedBy': ['http://purl.obolibrary.org/obo/obo.owl']}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Get ontology metadata\n",
    "for ont in adapter.ontologies():\n",
    "    ontology_metadata = adapter.ontology_metadata_map(ont)\n",
    "\n",
    "# Display ontology metadata\n",
    "display(ontology_metadata)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5443ff0c-34ba-409a-9382-be90da177816",
   "metadata": {},
   "source": [
    "### Load the ClinVar file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0f8d8136-70ed-4766-8f3f-4b805a4a2424",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene(s)</th>\n",
       "      <th>Protein change</th>\n",
       "      <th>Condition(s)</th>\n",
       "      <th>Accession</th>\n",
       "      <th>GRCh37Chromosome</th>\n",
       "      <th>GRCh37Location</th>\n",
       "      <th>GRCh38Chromosome</th>\n",
       "      <th>GRCh38Location</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>AlleleID(s)</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Canonical SPDI</th>\n",
       "      <th>Variant type</th>\n",
       "      <th>Molecular consequence</th>\n",
       "      <th>Germline classification</th>\n",
       "      <th>Germline date last evaluated</th>\n",
       "      <th>Germline review status</th>\n",
       "      <th>Somatic clinical impact</th>\n",
       "      <th>Somatic clinical impact date last evaluated</th>\n",
       "      <th>Somatic clinical impact review status</th>\n",
       "      <th>Oncogenicity classification</th>\n",
       "      <th>Oncogenicity date last evaluated</th>\n",
       "      <th>Oncogenicity review status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NM_001277115.2(DNAH11):c.6547-963G&gt;A</td>\n",
       "      <td>DNAH11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary ciliary dyskinesia 7</td>\n",
       "      <td>VCV003068449</td>\n",
       "      <td>7</td>\n",
       "      <td>21746354</td>\n",
       "      <td>7</td>\n",
       "      <td>21706736</td>\n",
       "      <td>3068449</td>\n",
       "      <td>2839546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000007.14:21706735:G:A</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>23-Jan-24</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NM_012381.4(ORC3):c.873+4A&gt;G</td>\n",
       "      <td>ORC3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ORC3-related disorder</td>\n",
       "      <td>VCV003068448</td>\n",
       "      <td>6</td>\n",
       "      <td>88321970</td>\n",
       "      <td>6</td>\n",
       "      <td>87612252</td>\n",
       "      <td>3068448</td>\n",
       "      <td>3227780</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000006.12:87612251:A:G</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>30-Aug-23</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Single allele</td>\n",
       "      <td>ADAM17|CPSF3|IAH1|LOC129933054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Neurodevelopmental disorder with microcephaly, hypotonia, nystagmus, and seizures</td>\n",
       "      <td>VCV003068447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>9444569 - 9551273</td>\n",
       "      <td>3068447</td>\n",
       "      <td>3227779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NM_000051.4(ATM):c.8152-260G&gt;A</td>\n",
       "      <td>ATM|C11orf65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "      <td>VCV003068446</td>\n",
       "      <td>11</td>\n",
       "      <td>108206312</td>\n",
       "      <td>11</td>\n",
       "      <td>108335585</td>\n",
       "      <td>3068446</td>\n",
       "      <td>3227778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000011.10:108335584:G:A</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_012434.5(SLC17A5):c.467_525+125del</td>\n",
       "      <td>SLC17A5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sialic acid storage disease, severe infantile type</td>\n",
       "      <td>VCV003068445</td>\n",
       "      <td>6</td>\n",
       "      <td>74351289 - 74351472</td>\n",
       "      <td>6</td>\n",
       "      <td>73641566 - 73641749</td>\n",
       "      <td>3068445</td>\n",
       "      <td>3227777</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>4-Apr-24</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Name                         Gene(s)  \\\n",
       "0   NM_001277115.2(DNAH11):c.6547-963G>A                          DNAH11   \n",
       "1           NM_012381.4(ORC3):c.873+4A>G                            ORC3   \n",
       "2                          Single allele  ADAM17|CPSF3|IAH1|LOC129933054   \n",
       "3         NM_000051.4(ATM):c.8152-260G>A                    ATM|C11orf65   \n",
       "4  NM_012434.5(SLC17A5):c.467_525+125del                         SLC17A5   \n",
       "\n",
       "  Protein change  \\\n",
       "0            NaN   \n",
       "1            NaN   \n",
       "2            NaN   \n",
       "3            NaN   \n",
       "4            NaN   \n",
       "\n",
       "                                                                        Condition(s)  \\\n",
       "0                                                       Primary ciliary dyskinesia 7   \n",
       "1                                                              ORC3-related disorder   \n",
       "2  Neurodevelopmental disorder with microcephaly, hypotonia, nystagmus, and seizures   \n",
       "3                                                     Ataxia-telangiectasia syndrome   \n",
       "4                                 Sialic acid storage disease, severe infantile type   \n",
       "\n",
       "      Accession GRCh37Chromosome       GRCh37Location GRCh38Chromosome  \\\n",
       "0  VCV003068449                7             21746354                7   \n",
       "1  VCV003068448                6             88321970                6   \n",
       "2  VCV003068447              NaN                  NaN                2   \n",
       "3  VCV003068446               11            108206312               11   \n",
       "4  VCV003068445                6  74351289 - 74351472                6   \n",
       "\n",
       "        GRCh38Location  VariationID AlleleID(s) dbSNP ID  \\\n",
       "0             21706736      3068449     2839546      NaN   \n",
       "1             87612252      3068448     3227780      NaN   \n",
       "2    9444569 - 9551273      3068447     3227779      NaN   \n",
       "3            108335585      3068446     3227778      NaN   \n",
       "4  73641566 - 73641749      3068445     3227777      NaN   \n",
       "\n",
       "               Canonical SPDI               Variant type  \\\n",
       "0   NC_000007.14:21706735:G:A  single nucleotide variant   \n",
       "1   NC_000006.12:87612251:A:G  single nucleotide variant   \n",
       "2                         NaN                   Deletion   \n",
       "3  NC_000011.10:108335584:G:A  single nucleotide variant   \n",
       "4                         NaN                   Deletion   \n",
       "\n",
       "  Molecular consequence Germline classification Germline date last evaluated  \\\n",
       "0        intron variant              Pathogenic                    23-Jan-24   \n",
       "1        intron variant       Likely pathogenic                    30-Aug-23   \n",
       "2                   NaN              Pathogenic                          NaN   \n",
       "3        intron variant  Uncertain significance                          NaN   \n",
       "4  splice donor variant       Likely pathogenic                     4-Apr-24   \n",
       "\n",
       "           Germline review status  Somatic clinical impact  \\\n",
       "0  no assertion criteria provided                      NaN   \n",
       "1  no assertion criteria provided                      NaN   \n",
       "2  no assertion criteria provided                      NaN   \n",
       "3  no assertion criteria provided                      NaN   \n",
       "4  no assertion criteria provided                      NaN   \n",
       "\n",
       "   Somatic clinical impact date last evaluated  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3                                          NaN   \n",
       "4                                          NaN   \n",
       "\n",
       "   Somatic clinical impact review status  Oncogenicity classification  \\\n",
       "0                                    NaN                          NaN   \n",
       "1                                    NaN                          NaN   \n",
       "2                                    NaN                          NaN   \n",
       "3                                    NaN                          NaN   \n",
       "4                                    NaN                          NaN   \n",
       "\n",
       "   Oncogenicity date last evaluated  Oncogenicity review status  \n",
       "0                               NaN                         NaN  \n",
       "1                               NaN                         NaN  \n",
       "2                               NaN                         NaN  \n",
       "3                               NaN                         NaN  \n",
       "4                               NaN                         NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('./data/input/clinvar_result.txt', delimiter='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bf3881d-b46f-4150-8d76-28b077bb1296",
   "metadata": {},
   "source": [
    "### Check Dataframe "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d611fd73-2aee-4b73-a461-2890e39ed1a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Name                                           737\n",
       "Gene(s)                                        537\n",
       "Protein change                                 538\n",
       "Condition(s)                                   659\n",
       "Accession                                      757\n",
       "GRCh37Chromosome                                25\n",
       "GRCh37Location                                 737\n",
       "GRCh38Chromosome                                25\n",
       "GRCh38Location                                 740\n",
       "VariationID                                    757\n",
       "AlleleID(s)                                    757\n",
       "dbSNP ID                                       604\n",
       "Canonical SPDI                                 685\n",
       "Variant type                                     8\n",
       "Molecular consequence                           66\n",
       "Germline classification                         12\n",
       "Germline date last evaluated                   528\n",
       "Germline review status                           6\n",
       "Somatic clinical impact                          0\n",
       "Somatic clinical impact date last evaluated      0\n",
       "Somatic clinical impact review status            0\n",
       "Oncogenicity classification                      0\n",
       "Oncogenicity date last evaluated                 0\n",
       "Oncogenicity review status                       0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.nunique()\n",
    "#Note: 659 unique Conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "90fb4a71-1b08-46bb-9037-eccb779d4559",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Condition(s)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Primary ciliary dyskinesia 7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ORC3-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neurodevelopmental disorder with microcephaly, hypotonia, nystagmus, and seizures</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sialic acid storage disease, severe infantile type</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        Condition(s)\n",
       "0                                                       Primary ciliary dyskinesia 7\n",
       "1                                                              ORC3-related disorder\n",
       "2  Neurodevelopmental disorder with microcephaly, hypotonia, nystagmus, and seizures\n",
       "3                                                     Ataxia-telangiectasia syndrome\n",
       "4                                 Sialic acid storage disease, severe infantile type"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_conditions_only = df[['Condition(s)']]\n",
    "df_conditions_only.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "baa50429-f3e6-480e-a18b-de8891132366",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Identify unique values in the 'Condition(s)' column\n",
    "# condition_counts = df_conditions_only['Condition(s)'].value_counts()\n",
    "# unique_conditions = condition_counts[condition_counts == 1].index\n",
    "\n",
    "# # Filter the DataFrame to include only rows with unique condition names\n",
    "# unique_conditions_df = df_conditions_only[df_conditions_only['Condition(s)'].isin(unique_conditions)]\n",
    "\n",
    "# unique_conditions_df.nunique() #Note: only 589 Conditions -> Incorrect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cb46a1b2-2a6d-46f2-a090-1b9e3da43e2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Identify unique values in the 'Condition(s)' column\n",
    "# unique_conditions = df_conditions_only['Condition(s)'].unique()\n",
    "\n",
    "# print(len(unique_conditions))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c8be32b4-ba18-46fc-a51c-696158bae051",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Condition(s)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Primary ciliary dyskinesia 7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ORC3-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Neurodevelopmental disorder with microcephaly, hypotonia, nystagmus, and seizures</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sialic acid storage disease, severe infantile type</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        Condition(s)\n",
       "0                                                       Primary ciliary dyskinesia 7\n",
       "1                                                              ORC3-related disorder\n",
       "2  Neurodevelopmental disorder with microcephaly, hypotonia, nystagmus, and seizures\n",
       "3                                                     Ataxia-telangiectasia syndrome\n",
       "4                                 Sialic acid storage disease, severe infantile type"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop duplicates to keep only unique conditions\n",
    "unique_conditions_df = df_conditions_only.drop_duplicates(subset='Condition(s)', keep='first')\n",
    "\n",
    "# Optionally, filter out empty values\n",
    "# unique_conditions_df = unique_conditions_df[unique_conditions_df['Condition(s)'].str.strip() != '']\n",
    "\n",
    "# Reset the index for better readability\n",
    "unique_conditions_df = unique_conditions_df.reset_index(drop=True)\n",
    "\n",
    "# Display the filtered DataFrame\n",
    "unique_conditions_df.nunique()\n",
    "\n",
    "unique_conditions_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c33af89c-48a7-40dc-a847-16d0aeb92db0",
   "metadata": {},
   "source": [
    "### Search the Unique Conditions to see if they exist in Mondo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7a317cb0-ea75-46c0-a48c-bb2bc9d37c09",
   "metadata": {},
   "outputs": [],
   "source": [
    "# See https://incatools.github.io/ontology-access-kit/intro/tutorial01.html#search\n",
    "# NOTE: Search is case-sensitive\n",
    "\n",
    "# !runoak --input sqlite:obo:mondo search 'primary ciliary dyskinesia 7' # prefLabel\n",
    "# !runoak --input sqlite:obo:mondo search 'ciliary dyskinesia, primary, type 7' #exactSynonym\n",
    "# !runoak --input sqlite:obo:mondo search 'l~PRimary ciliary dyskinesia 7' #this is case insensitive, but only for label matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "118d0a07-4dbb-472f-baf8-593570600c79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Search all unique conditions from the ClinVar UDN submission\n",
    "# tqdm.pandas()\n",
    "\n",
    "# def run_oak_search(condition):\n",
    "#     command = f\"runoak --input sqlite:obo:mondo search '{condition}'\"\n",
    "#     result = !{command}\n",
    "#     return '\\n'.join(result)\n",
    "\n",
    "# # Apply the function to each row in the 'Condition(s)' column and store the result in a new column\n",
    "# unique_conditions_df['search_result'] = unique_conditions_df['Condition(s)'].progress_apply(run_oak_search)\n",
    "\n",
    "# # View results\n",
    "# unique_conditions_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "15ae4092-5ba5-4a59-87e2-2648bcf2befe",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write results to file\n",
    "unique_conditions_df.to_csv('./data/output/unique_conditions_df', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5de9b8d4-4dc7-42c7-95ea-6f960da38d17",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Filter the DataFrame to get rows where 'search_result' is empty\n",
    "# empty_results_df = unique_conditions_df[unique_conditions_df['search_result'].str.strip() == '']\n",
    "\n",
    "# empty_results_df.nunique() #Condition(s)     583"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "62f2829b-be43-4ad6-82c9-aec2d8cddef0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Filter the DataFrame to get rows where 'search_result' has a value (is not empty)\n",
    "# non_empty_results_df = unique_conditions_df[unique_conditions_df['search_result'].str.strip() != '']\n",
    "# non_empty_results_df.nunique() #Condition(s)     76\n",
    "\n",
    "# Save the filtered DataFrame to a CSV file\n",
    "# non_empty_results_df.to_csv('non_empty_search_results.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f651384-9666-4350-a834-3abfe43d1ea1",
   "metadata": {},
   "source": [
    "---\n",
    "### Search Issues\n",
    "- ClinVar does not use Mondo labels for the disease name, e.g. Ataxia-telangiectasia syndrome is actually Ataxia telangiectasia syndrome. \n",
    "- OAK Search is case sensitive\n",
    "\n",
    "Use another data source that has the ClinVar disease name and Mondo ID to see which ClinVar UDN submissions "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "934b09e4-62cd-4c93-ac27-cc2cd563c147",
   "metadata": {},
   "source": [
    "### Use another data source file - MedGenIDMappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "de752993-ea27-4053-89e5-48febffec7db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2024-06-15 00:23:13--  https://ftp.ncbi.nlm.nih.gov/pub/medgen/MedGenIDMappings.txt.gz\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 2607:f220:41e:250::12, 2607:f220:41e:250::13, 130.14.250.7, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|2607:f220:41e:250::12|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 5636361 (5.4M) [application/x-gzip]\n",
      "Saving to: ‘./data/tmp/MedGenIDMappings.txt.gz’\n",
      "\n",
      "./data/tmp/MedGenID 100%[===================>]   5.38M  6.47MB/s    in 0.8s    \n",
      "\n",
      "2024-06-15 00:23:14 (6.47 MB/s) - ‘./data/tmp/MedGenIDMappings.txt.gz’ saved [5636361/5636361]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Try to use MedGenIDMappings.txt from https://ftp.ncbi.nlm.nih.gov/pub/medgen/\n",
    "\n",
    "# Fetch file\n",
    "!wget -O ./data/tmp/MedGenIDMappings.txt.gz https://ftp.ncbi.nlm.nih.gov/pub/medgen/MedGenIDMappings.txt.gz\n",
    "\n",
    "# Unzip\n",
    "!gunzip -f ./data/tmp/MedGenIDMappings.txt.gz #Force overwriting any existing file named MedGenIDMappings.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1c21b47b-0400-45ba-bc08-e5cdebca70e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CUI_or_CN_id</th>\n",
       "      <th>pref_name</th>\n",
       "      <th>source_id</th>\n",
       "      <th>source</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C0000727</td>\n",
       "      <td>Acute abdomen</td>\n",
       "      <td>9209005</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C0000727</td>\n",
       "      <td>Acute abdomen</td>\n",
       "      <td>HP:0033400</td>\n",
       "      <td>HPO</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C0000731</td>\n",
       "      <td>Abdominal distention</td>\n",
       "      <td>34</td>\n",
       "      <td>MedGen</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C0000734</td>\n",
       "      <td>Abdominal mass</td>\n",
       "      <td>271860004</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C0000735</td>\n",
       "      <td>Neoplasm of abdomen</td>\n",
       "      <td>128050000</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CUI_or_CN_id             pref_name   source_id       source  Unnamed: 4\n",
       "0      C0000727         Acute abdomen     9209005  SNOMEDCT_US         NaN\n",
       "1      C0000727         Acute abdomen  HP:0033400          HPO         NaN\n",
       "2      C0000731  Abdominal distention          34       MedGen         NaN\n",
       "3      C0000734        Abdominal mass   271860004  SNOMEDCT_US         NaN\n",
       "4      C0000735   Neoplasm of abdomen   128050000  SNOMEDCT_US         NaN"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in MedGenIDMappings file\n",
    "medgen_df = pd.read_csv('./data/tmp/MedGenIDMappings.txt', delimiter='|')\n",
    "medgen_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7b2055a3-9ce4-45f7-bf17-343325402497",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CUI_or_CN_id</th>\n",
       "      <th>pref_name</th>\n",
       "      <th>source_id</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C0000727</td>\n",
       "      <td>Acute abdomen</td>\n",
       "      <td>9209005</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C0000727</td>\n",
       "      <td>Acute abdomen</td>\n",
       "      <td>HP:0033400</td>\n",
       "      <td>HPO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C0000731</td>\n",
       "      <td>Abdominal distention</td>\n",
       "      <td>34</td>\n",
       "      <td>MedGen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C0000734</td>\n",
       "      <td>Abdominal mass</td>\n",
       "      <td>271860004</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C0000735</td>\n",
       "      <td>Neoplasm of abdomen</td>\n",
       "      <td>128050000</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CUI_or_CN_id             pref_name   source_id       source\n",
       "0      C0000727         Acute abdomen     9209005  SNOMEDCT_US\n",
       "1      C0000727         Acute abdomen  HP:0033400          HPO\n",
       "2      C0000731  Abdominal distention          34       MedGen\n",
       "3      C0000734        Abdominal mass   271860004  SNOMEDCT_US\n",
       "4      C0000735   Neoplasm of abdomen   128050000  SNOMEDCT_US"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop empty last column\n",
    "medgen_df = medgen_df.drop(columns=['Unnamed: 4'])\n",
    "medgen_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a60e8a82-d71d-40bc-8e34-d19e238e8d93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Condition(s)    659\n",
       "dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_conditions_df.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1c007b1b-c63e-4285-b836-19f177633cfc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty DataFrame\n",
      "Columns: [#CUI_or_CN_id, pref_name, source_id, source]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "search_value = 'Neurodevelopmental delay|Seizure'\n",
    "\n",
    "result = medgen_df[medgen_df['pref_name'] == search_value]\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91c9c4cc-ab8c-42a6-af73-1f24be4efa41",
   "metadata": {},
   "source": [
    "### Find ClinVar UDN submissions with Mondo IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4d05dc3e-3ad1-4eb0-9b4e-5aab3eb8b8f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CUI_or_CN_id</th>\n",
       "      <th>pref_name</th>\n",
       "      <th>source_id</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C0004135</td>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "      <td>208900</td>\n",
       "      <td>OMIM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C0004135</td>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "      <td>208900</td>\n",
       "      <td>OMIM included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C0004135</td>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "      <td>68504005</td>\n",
       "      <td>SNOMEDCT_US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C0004135</td>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "      <td>439</td>\n",
       "      <td>MedGen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C0004135</td>\n",
       "      <td>Ataxia-telangiectasia syndrome</td>\n",
       "      <td>5862</td>\n",
       "      <td>GARD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #CUI_or_CN_id                       pref_name source_id         source\n",
       "0      C0004135  Ataxia-telangiectasia syndrome    208900           OMIM\n",
       "1      C0004135  Ataxia-telangiectasia syndrome    208900  OMIM included\n",
       "2      C0004135  Ataxia-telangiectasia syndrome  68504005    SNOMEDCT_US\n",
       "3      C0004135  Ataxia-telangiectasia syndrome       439         MedGen\n",
       "4      C0004135  Ataxia-telangiectasia syndrome      5862           GARD"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Join medgen_df to original ClinVar UDN submission file with only unique Conditions, e.g. unique_conditions_df\n",
    "\n",
    "# Ensure the columns you want to join on have the same column name\n",
    "unique_conditions_df = unique_conditions_df.rename(columns={'Condition(s)': 'pref_name'})\n",
    "\n",
    "# Perform the merge\n",
    "clinvar_mondo_df = pd.merge(medgen_df, unique_conditions_df, on='pref_name', how='inner')\n",
    "\n",
    "clinvar_mondo_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "927eb72c-1464-4141-87da-5d43f10ed1cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "#CUI_or_CN_id     237\n",
       "pref_name         237\n",
       "source_id        1708\n",
       "source             11\n",
       "dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_mondo_df.nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44541269-5356-4774-a290-0be5ed00f0ef",
   "metadata": {},
   "source": [
    "### Find ClinVar UDN submissions _without_ Mondo IDs\n",
    "\n",
    "Find the rows from the unique list of Conditions from the ClinVar UDN submission that do not have a Mondo ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6db7ffa5-1c16-47a1-836d-144b5c2688fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pref_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ORC3-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>ADCY2-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507</th>\n",
       "      <td>ZNF331 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>NOTCH1-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>KDR-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>KIF5B-related osteogenesis imperfecta syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>PAPSS1-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545</th>\n",
       "      <td>MYCBP2-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>TMEM161B-related lissencephaly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>547</th>\n",
       "      <td>DHX9-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>548</th>\n",
       "      <td>DAGLA-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>555</th>\n",
       "      <td>SPTSSA-related disorder|Spastic paraplegia 90A, autosomal dominant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>625</th>\n",
       "      <td>ZNF865-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>626</th>\n",
       "      <td>SREBF2-related disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>643</th>\n",
       "      <td>Neurodevelopmental delay|Seizure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>683</th>\n",
       "      <td>Spastic ataxia 5|not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>RRAGD-related disorder|Hypomagnesemia 7, renal, with or without dilated cardiomyopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>not provided|PLA2G6-associated neurodegeneration</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>DPH5-related diphthamide-deficiency syndrome|Neurodevelopmental disorder with short stature, prominent forehead, and feeding difficulties</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>699</th>\n",
       "      <td>IQGAP1-associated immune condition</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                     pref_name\n",
       "5                                                                                                                        ORC3-related disorder\n",
       "70                                                                                                                      ADCY2-related disorder\n",
       "507                                                                                                                            ZNF331 deletion\n",
       "514                                                                                                                    NOTCH1-related disorder\n",
       "518                                                                                                                       KDR-related disorder\n",
       "539                                                                                             KIF5B-related osteogenesis imperfecta syndrome\n",
       "540                                                                                                                    PAPSS1-related disorder\n",
       "545                                                                                                                    MYCBP2-related disorder\n",
       "546                                                                                                             TMEM161B-related lissencephaly\n",
       "547                                                                                                                      DHX9-related disorder\n",
       "548                                                                                                                     DAGLA-related disorder\n",
       "555                                                                         SPTSSA-related disorder|Spastic paraplegia 90A, autosomal dominant\n",
       "625                                                                                                                    ZNF865-related disorder\n",
       "626                                                                                                                    SREBF2-related disorder\n",
       "643                                                                                                           Neurodevelopmental delay|Seizure\n",
       "683                                                                                                              Spastic ataxia 5|not provided\n",
       "693                                                     RRAGD-related disorder|Hypomagnesemia 7, renal, with or without dilated cardiomyopathy\n",
       "694                                                                                           not provided|PLA2G6-associated neurodegeneration\n",
       "698  DPH5-related diphthamide-deficiency syndrome|Neurodevelopmental disorder with short stature, prominent forehead, and feeding difficulties\n",
       "699                                                                                                         IQGAP1-associated immune condition"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform a left join to find all rows from unique_conditions_df (ClinVar UDN submission) and their matches in medgen_df\n",
    "merged_df = pd.merge(unique_conditions_df, medgen_df, on='pref_name', how='left', indicator=True)\n",
    "# merged_df.nunique()\n",
    "\n",
    "# Filter to find rows that were not merged (i.e., rows from unique_conditions_df with no match in medgen_df)\n",
    "clinvar_no_mondo_df = merged_df[merged_df['_merge'] == 'left_only']\n",
    "\n",
    "# Drop the merge indicator column\n",
    "clinvar_no_mondo_df = clinvar_no_mondo_df.drop(columns=['_merge', '#CUI_or_CN_id', 'source_id', 'source'])\n",
    "\n",
    "clinvar_no_mondo_df.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "0c692b14-e553-4720-8a8f-c9b8b4eb38f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pref_name    422\n",
       "dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_no_mondo_df.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2d747621-472d-4625-9a2b-f1725c968d74",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_no_mondo_df.to_csv('./data/output/clinvar_no_mondo.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34a074e3-69c4-4e80-b66e-2e9416ba26ca",
   "metadata": {},
   "source": [
    "---\n",
    "### Compare ClinVar UDN submissions to MedGen file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c99623b6-e040-4c45-9104-6a908484fe17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Submissions</th>\n",
       "      <th>Last Updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1p13.3 deletion syndrome</td>\n",
       "      <td>1</td>\n",
       "      <td>2019-06-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3-methylcrotonyl-CoA carboxylase 2 deficiency</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-01-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>46,XX sex reversal 4</td>\n",
       "      <td>1</td>\n",
       "      <td>2017-08-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA2-related disorder</td>\n",
       "      <td>2</td>\n",
       "      <td>2018-03-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACOX1-related disorder</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-01-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Name  Submissions Last Updated\n",
       "0                       1p13.3 deletion syndrome            1   2019-06-24\n",
       "1  3-methylcrotonyl-CoA carboxylase 2 deficiency            1   2024-01-19\n",
       "2                           46,XX sex reversal 4            1   2017-08-04\n",
       "3                         ABCA2-related disorder            2   2018-03-27\n",
       "4                         ACOX1-related disorder            1   2024-01-03"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_udn_df = pd.read_excel('./data/input/clinvar_udn_submissions.xlsx')\n",
    "\n",
    "clinvar_udn_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e3f66814-d3d5-4d08-80d3-b4ebe1bcbcd2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Name            539\n",
       "Submissions       6\n",
       "Last Updated     47\n",
       "dtype: int64"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_udn_df.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "94ace540-5446-481a-bc42-fd63bb073d7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Submissions</th>\n",
       "      <th>Last Updated</th>\n",
       "      <th>#CUI_or_CN_id</th>\n",
       "      <th>pref_name</th>\n",
       "      <th>source_id</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3-methylcrotonyl-CoA carboxylase 2 deficiency</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-01-19</td>\n",
       "      <td>C1859499</td>\n",
       "      <td>3-methylcrotonyl-CoA carboxylase 2 deficiency</td>\n",
       "      <td>MONDO:0008862</td>\n",
       "      <td>MONDO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46,XX sex reversal 4</td>\n",
       "      <td>1</td>\n",
       "      <td>2017-08-04</td>\n",
       "      <td>C4479552</td>\n",
       "      <td>46,XX sex reversal 4</td>\n",
       "      <td>MONDO:0060489</td>\n",
       "      <td>MONDO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADNP-related multiple congenital anomalies - intellectual disability - autism spectrum disorder</td>\n",
       "      <td>2</td>\n",
       "      <td>2020-04-23</td>\n",
       "      <td>C4014538</td>\n",
       "      <td>ADNP-related multiple congenital anomalies - intellectual disability - autism spectrum disorder</td>\n",
       "      <td>MONDO:0014379</td>\n",
       "      <td>MONDO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AHDC1-related intellectual disability - obstructive sleep apnea - mild dysmorphism syndrome</td>\n",
       "      <td>1</td>\n",
       "      <td>2021-12-03</td>\n",
       "      <td>C4014419</td>\n",
       "      <td>AHDC1-related intellectual disability - obstructive sleep apnea - mild dysmorphism syndrome</td>\n",
       "      <td>MONDO:0014358</td>\n",
       "      <td>MONDO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ALG1-congenital disorder of glycosylation</td>\n",
       "      <td>3</td>\n",
       "      <td>2018-03-27</td>\n",
       "      <td>C2931005</td>\n",
       "      <td>ALG1-congenital disorder of glycosylation</td>\n",
       "      <td>MONDO:0012052</td>\n",
       "      <td>MONDO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                              Name  \\\n",
       "0                                                    3-methylcrotonyl-CoA carboxylase 2 deficiency   \n",
       "1                                                                             46,XX sex reversal 4   \n",
       "2  ADNP-related multiple congenital anomalies - intellectual disability - autism spectrum disorder   \n",
       "3      AHDC1-related intellectual disability - obstructive sleep apnea - mild dysmorphism syndrome   \n",
       "4                                                        ALG1-congenital disorder of glycosylation   \n",
       "\n",
       "   Submissions Last Updated #CUI_or_CN_id  \\\n",
       "0            1   2024-01-19      C1859499   \n",
       "1            1   2017-08-04      C4479552   \n",
       "2            2   2020-04-23      C4014538   \n",
       "3            1   2021-12-03      C4014419   \n",
       "4            3   2018-03-27      C2931005   \n",
       "\n",
       "                                                                                         pref_name  \\\n",
       "0                                                    3-methylcrotonyl-CoA carboxylase 2 deficiency   \n",
       "1                                                                             46,XX sex reversal 4   \n",
       "2  ADNP-related multiple congenital anomalies - intellectual disability - autism spectrum disorder   \n",
       "3      AHDC1-related intellectual disability - obstructive sleep apnea - mild dysmorphism syndrome   \n",
       "4                                                        ALG1-congenital disorder of glycosylation   \n",
       "\n",
       "       source_id source  \n",
       "0  MONDO:0008862  MONDO  \n",
       "1  MONDO:0060489  MONDO  \n",
       "2  MONDO:0014379  MONDO  \n",
       "3  MONDO:0014358  MONDO  \n",
       "4  MONDO:0012052  MONDO  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### How many entries in clinvar_udn_submissions.xlsx are a disease in Mondo?\n",
    "\n",
    "# Filter medgen_df to include only rows where 'source' is 'MONDO'\n",
    "filtered_medgen_df = medgen_df[medgen_df['source'] == 'MONDO']\n",
    "\n",
    "# Merge the DataFrames on the columns of interest to find matches\n",
    "merged_mondo_df = pd.merge(clinvar_udn_df, filtered_medgen_df, left_on='Name', right_on='pref_name', how='inner')\n",
    "\n",
    "merged_mondo_df.head()\n",
    "\n",
    "# Count the number of matching rows\n",
    "# matching_row_count = merged_mondo_df.shape[0]\n",
    "# print(matching_row_count) # 395"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "5a99c483-13e2-4035-9cef-9b3f9a099c7a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Submissions</th>\n",
       "      <th>Last Updated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1p13.3 deletion syndrome</td>\n",
       "      <td>1</td>\n",
       "      <td>2019-06-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABCA2-related disorder</td>\n",
       "      <td>2</td>\n",
       "      <td>2018-03-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACOX1-related disorder</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-01-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ADCY2-related disorder</td>\n",
       "      <td>1</td>\n",
       "      <td>2024-04-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ADGRV1-related myoclonic epilepsy</td>\n",
       "      <td>1</td>\n",
       "      <td>2019-09-17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Name  Submissions Last Updated\n",
       "0           1p13.3 deletion syndrome            1   2019-06-24\n",
       "3             ABCA2-related disorder            2   2018-03-27\n",
       "4             ACOX1-related disorder            1   2024-01-03\n",
       "5             ADCY2-related disorder            1   2024-04-05\n",
       "6  ADGRV1-related myoclonic epilepsy            1   2019-09-17"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### How many entries in clinvar_udn_submissions.xlsx are a _not_ disease in Mondo?\n",
    "\n",
    "# Perform a left join to keep all rows from clinvar_udn_df and add matching rows from filtered_medgen_df\n",
    "merged_not_mondo_df = pd.merge(clinvar_udn_df, filtered_medgen_df, left_on='Name', right_on='pref_name', how='left', indicator=True)\n",
    "\n",
    "# Filter the DataFrame to keep only rows where 'pref_name' is NaN (no match found)\n",
    "non_matching_rows = merged_not_mondo_df[merged_not_mondo_df['_merge'] == 'left_only']\n",
    "\n",
    "# Select relevant columns (dropping merge indicator columns and columns from filtered_medgen_df)\n",
    "non_matching_rows = non_matching_rows.drop(columns=['pref_name', 'source', '_merge', '#CUI_or_CN_id', 'source_id'])\n",
    "\n",
    "# Display the non-matching rows\n",
    "non_matching_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "7301b1db-bf5d-40e6-8c7d-23fa52e1d5e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Name            144\n",
       "Submissions       5\n",
       "Last Updated     34\n",
       "dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_matching_rows.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "07d1bd8a-92bf-4a90-8560-9744a485fc29",
   "metadata": {},
   "outputs": [],
   "source": [
    "non_matching_rows.to_csv('./data/output/clinvar_udn_submissions-no_mondo.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d34b458-e362-4d49-bc9b-f30f294d12b8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
